wd_helper.au3

```diff
@@ -1247,155 +1249,165 @@ EndFunc   ;==>_WD_IsLatestRelease
 	Local Const $sFuncName = "_WD_UpdateDriver"
 	Local $iErr = $_WD_ERROR_Success, $sDriverEXE, $sBrowserVersion, $bResult = False
-	Local $sDriverVersion, $sVersionShort, $sDriverLatest, $sURLNewDriver
-	Local $sReturned, $sTempFile, $hFile, $oShell, $FilesInZip, $sResult, $iStartPos, $iConversion
+	Local $sDriverCurrent, $sVersionShort, $sDriverLatest, $sURLNewDriver
+	Local $sTempFile, $oShell, $FilesInZip, $sResult, $iStartPos, $iConversion
 
 	If $sInstallDir = Default Then $sInstallDir = @ScriptDir
+	$sInstallDir = StringRegExpReplace($sInstallDir, '(?i)(\\)\Z', '') & '\' ; prevent double \\ on the end of directory
 	If $bFlag64 = Default Then $bFlag64 = False
 	If $bForce = Default Then $bForce = False
 
 	; If the Install directory doesn't exist and it can't be created, then set error
-	If (Not FileExists($sInstallDir)) And (Not DirCreate($sInstallDir)) Then $iErr = $_WD_ERROR_InvalidValue
-
-	; Save current debug level and set to none
-	Local $WDDebugSave = $_WD_DEBUG
-	$_WD_DEBUG = $_WD_DEBUG_None
-
-	Switch $sBrowser
-		Case 'chrome'
-			$sDriverEXE = "chromedriver.exe"
-		Case 'firefox'
-			$sDriverEXE = "geckodriver.exe"
-		Case 'msedge'
-			$sDriverEXE = "msedgedriver.exe"
-		Case Else
-			$iErr = $_WD_ERROR_InvalidValue
-	EndSwitch
+	If (Not FileExists($sInstallDir)) And (Not DirCreate($sInstallDir)) Then
+		$iErr = $_WD_ERROR_InvalidValue
+	Else
+		; Save current debug level and set to none
+		Local $WDDebugSave = $_WD_DEBUG
+		$_WD_DEBUG = $_WD_DEBUG_None
 
-	If $iErr = $_WD_ERROR_Success Then
 		$sBrowserVersion = _WD_GetBrowserVersion($sBrowser)
-		If @error Then $iErr = @error
+		$iErr = @error
+		If $iErr = $_WD_ERROR_Success Then
+			Switch $sBrowser
+				Case 'chrome'
+					$sDriverEXE = "chromedriver.exe"
+				Case 'firefox'
+					$sDriverEXE = "geckodriver.exe"
+				Case 'msedge'
+					$sDriverEXE = "msedgedriver.exe"
+			EndSwitch
 
-		$sDriverVersion = _WD_GetWebDriverVersion($sInstallDir, $sDriverEXE)
-		If @error Then $iErr = @error
+			; Determine current local webdriver Architecture
+			If FileExists($sInstallDir & $sDriverEXE) Then
+				_WinAPI_GetBinaryType($sInstallDir & $sDriverEXE)
+				Local $bDriverIs64Bit = (@extended = $SCS_64BIT_BINARY)
+				If $sBrowser <> 'chrome' Then
+					If $WDDebugSave = $_WD_DEBUG_Info Then
+						__WD_ConsoleWrite($sFuncName & ': ' & $sDriverEXE & ' = ' & (($bDriverIs64Bit) ? ("switching 64>32 Bit") : ("switching 32>64 Bit")) & @CRLF)
+					EndIf
 
-		; Determine latest available webdriver version
-		; for the designated browser
-		Switch $sBrowser
-			Case 'chrome'
-				$sVersionShort = StringLeft($sBrowserVersion, StringInStr($sBrowserVersion, ".", 0, -1) - 1)
-				$sDriverLatest = BinaryToString(InetRead('https://chromedriver.storage.googleapis.com/LATEST_RELEASE_' & $sVersionShort))
-				$sURLNewDriver = "https://chromedriver.storage.googleapis.com/" & $sDriverLatest & "/chromedriver_win32.zip"
+					If $bDriverIs64Bit <> $bFlag64 Then FileDelete($sInstallDir & $sDriverEXE)
```

Comment:
Do we really want to delete the driver here? There's existing code that will delete the driver, so I think you could just set `$bForce = True` here.

Reply:
Done